homeassistant/components/shelly/entity.py

```diff
@@ -472,6 +536,58 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
         return self.description.extra_state_attributes(self.wrapper.device.status)
 
 
+class ShellyRpcAttributeEntity(ShellyRpcEntity, entity.Entity):
+    """Helper class to represent a rpc attribute."""
+
+    def __init__(
+        self,
+        wrapper: RpcDeviceWrapper,
+        key: str,
+        attribute: str,
+        description: RpcAttributeDescription,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(wrapper, key)
+        self.attribute = attribute
+        self.description = description
+
+        self._attr_unique_id = f"{super().unique_id}-{attribute}"
+        self._attr_name = get_rpc_entity_name(wrapper.device, key, description.name)
+        self._attr_entity_registry_enabled_default = description.default_enabled
+        self._attr_device_class = description.device_class
+        self._attr_icon = description.icon
+        self._last_value = None
+
+    @property
+    def attribute_value(self) -> StateType:
+        """Value of sensor."""
+        if callable(self.description.value):
+            self._last_value = self.description.value(
+                self.wrapper.device.status[self.key], self._last_value
+            )
+        return self._last_value
+
+    @property
+    def available(self) -> bool:
```

Comment:
Can be swithed to` self._attr_available` ?

Reply:
If we want to switch to `self.__attr_available`  we must do it inside an update method which I don't see how that would be any improvement to code quality.
In my opinion `_attr_` types should only happen for attributes which are never changed after entity creation, this is too much abused today in core, the idea behind the `_attr_` types was to improve code quality and for static attributes is has great improvement, but for attributes which are changed on updated I think a property is a better solution.